src/res.rs

```diff
@@ -263,36 +274,67 @@ impl Resources {
     pub fn fetch_mut<T>(&self, id: usize) -> FetchMut<T>
         where T: Resource
     {
-        let c = self.fetch_internal(TypeId::of::<T>(), id);
+        self.try_fetch_mut(id).expect(RESOURCE_NOT_FOUND)
+    }
 
-        FetchMut {
-            inner: c.borrow_mut(),
-            phantom: PhantomData,
-        }
+    /// Like `fetch_mut`, but returns an `Option` instead of panicking in the case of the resource
+    /// being accessed mutably.
+    pub fn try_fetch_mut<T>(&self, id: usize) -> Option<FetchMut<T>>
+        where T: Resource
+    {
+        self.try_fetch_internal(TypeId::of::<T>(), id)
+            .map(|r| {
+            FetchMut {
+                inner: r.borrow_mut(),
+                phantom: PhantomData,
+            }
+        })
     }
 
     /// Fetches the resource with the specified type id.
     ///
     /// Please see `fetch` for details.
+    ///
+    /// # Panics
+    ///
+    /// Panics if no resource with the id exists.
     pub fn fetch_id(&self, id: TypeId, comp_id: usize) -> FetchId {
```

Comment:
Last thing: I think we should stay consistent with `fetch` then, let's use `try_fetch` here and `.expect` the value.

We should be able to get rid of `fetch_internal` then.